server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -688,12 +692,14 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
             } else {
                 o.writeByte((byte) 10);
             }
-            @SuppressWarnings("unchecked")
-            final Map<String, Object> map = (Map<String, Object>) v;
-            o.writeVInt(map.size());
-            for (Map.Entry<String, Object> entry : map.entrySet()) {
-                o.writeString(entry.getKey());
-                o.writeGenericValue(entry.getValue());
+            if (o.getVersion().onOrAfter(Version.V_8_7_0)) {
+                @SuppressWarnings("unchecked")
+                final Map<Object, Object> map = (Map<Object, Object>) v;
```

Comment:
maybe `final Map<?, ?> map = (Map<?, ?>) v;` ? Then `@SuppressWarnings("unchecked")` can be removed.